infrastructure/src/main/java/org/corfudb/infrastructure/ServerHandshakeHandler.java

```diff
@@ -0,0 +1,213 @@
+package org.corfudb.infrastructure;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.channel.group.ChannelGroup;
+import io.netty.util.AttributeKey;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.corfudb.protocols.wireprotocol.CorfuMsg;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.HandshakeMsg;
+import org.corfudb.protocols.wireprotocol.HandshakeResponse;
+
+/**
+ * The ServerHandshakeHandler waits for the handshake message, validates and sends
+ * a response to the client. This reply contains its node id and current version of Corfu.
+ *
+ * Created by amartinezman on 12/11/17.
+ */
+@Slf4j
+@ChannelHandler.Sharable
```

Comment:
Is this really Sharable? handshakeComplete looks like state.

Reply:
Totally agree, I thought I had removed it. It is not sharable.